source/core/ExecutionContext.cpp

```diff
@@ -198,6 +198,41 @@ VIREO_FUNCTION_SIGNATURE1(Branch, InstructionCore)
     return _ParamPointer(0);
 }
 //------------------------------------------------------------
+VIREO_FUNCTION_SIGNATURE1(CallChain, StringRefArray1D*)
+{
+    ExecutionContextRef exec = THREAD_EXEC();
+
+    VIClump* runningQueueElt = exec->_runningQueueElt;
+    VirtualInstrument* vi = runningQueueElt->OwningVI();
+    VIClump *caller = runningQueueElt;
+    Int32 count = 0;
+
+    if (_ParamPointer(0)) {
+        do { // preflight caller chain to count subVI depth
+            caller = vi->Clumps()->Begin()->_caller; // caller only set on entry clump
+            if (caller)
+                vi = caller->OwningVI();
+            ++count;
+        } while (caller);
+
+        _Param(0)->Resize1D(count);
+
+        count = 0;
+        vi = runningQueueElt->OwningVI();
+        do {
+            SubString s = vi->VIName();
+            _Param(0)->At(count)->CopyFromSubString(&s);
```

Comment:
Is it a good idea to make sure "count" is less than what we have allocated before from the previous while loop to avoid any potential access violation errors.